doc/script_commands.txt

```diff
@@ -6611,18 +6611,49 @@ Examples:
 *unitwarp(<GID>, <Mapname>, <x>, <y>)
 *unitattack(<GID>, <Target ID>)
 *unitstop(<GID>)
-*unittalk(<GID>, <Text>)
+*unittalk(<GID>, <Text>{, show_name{, <send_target>{, <target_id>}}})
 *unitemote(<GID>, <Emote>)
 
 Okay, these commands should be fairly self explaining.
 For the emotions, you can look in db/constants.conf for prefixes with e_
 PS: unitwarp() supports a <GID> of zero, which causes the executor of the
+PS: unittalk() can receive 3 extra parameters:
+	show_name:
+		true: Shows Unit name like "UnitName : Message" (default)
+		false: Hides Unit name
+	send_target:
+		AREA_CHAT_WOC: sends the message to everyone in view range including the attached unit (default)
+		SELF: sends the message to the given unit gid only
+	target_id:
+		if send_target is set to SELF, the message will be shown by the given gid,
+		target_id allows changing the unit that will see the message.
 script to be affected. This can be used with OnTouchNPC to warp
 monsters:
 
 OnTouchNPC:
 	unitwarp(0, "this", -1, -1);
 
+	// hide the npc name from the text
+	unittalk(getnpcid(0), "foobar", false);
+
+	// display by npc to everyone
+	unittalk(getnpcid(0), "foobar", true);
+
+	// display by npc to npc
+	unittalk(getnpcid(0), "foobar", true, SELF);
```

Comment:
have you try this line in-game ?
what does it mean by `display by npc to npc` ?
test in-game and it does nothing

also tried defpattern, npctalk doesn't trigger the defpattern
because it only run when player talk to npc, so what's the point of having npc talk to itself ?

Reply:
This will send the message where only the npc itself can see it, this is useless but for now I didn't want to restrict the command and kept the behavior the same across all unit types however I would like to see @MishimaHaruna input on this, should it be restricted to BL_PC only or it's fine like this.

Reply:
I'm also proposing the format 
*unittalk(<GID>, <Text>{, <flag>{, <account ID>}})

old behavior, send to all with npc name
unittalk getnpcid(0), "blah";

send to all without npc name
unittalk getnpcid(0), "blah", false;

send the chat message from npc to only 1 player only
unittalk getnpcid(0), "blah", true, getcharid(3);

not really sure why you want the monster/pet/npc to hear them
unless we are going to change defpattern to allow it trigger from non-player sources ?

Reply:
Is this "by npc to npc" still true in the current version of the PR? Wouldn't it send the message to the attached player instead? (I think defaulting to the attached player instead of the specified GID, would cover more use cases)

Am I misreading it, or this is already handled here?
```
		if (target == SELF && (target_bl == NULL || bl == target_bl)) {
			clif->notify_playerchat(bl, StrBuf->Value(&sbuf));
		} else {
			clif->disp_overhead(bl, StrBuf->Value(&sbuf), target, target_bl);
		} 
```

Reply:
@MishimaHaruna actually in current version this is a mistake `clif->notify_playerchat` is a player only packet and should not cover the case of other units.